### PR TITLE
[luci/lang] Introduce DataTypeHelper

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce DataTypeHelper with size() of DataType.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
for #12717 , from #12743

**Comment by seanshpark** [comment]
need rebase after #12754 lands

File: compiler/luci/lang/include/luci/IR/DataTypeHelper.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+#include <loco/IR/DataType.h>
+#include <loco/IR/DataTypeTraits.h>
+
+namespace luci
+{
+
+/**
+ * @brief Returns the size of the data type.
+ * @note  luci will use S4, U4 as byte
```
Comment (jinevening) [inline compiler/luci/lang/include/luci/IR/DataTypeHelper.h:28]: To clarify, how about adding more explanation?

```suggestion
 * @note  luci saves S4, U4 in a single byte. The extra 4 bits in the MSB side are filled with sign bits.
```

**Review by jinevening — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
> need rebase after #12754 lands

current change is header only...

I'll add dummy cpp in another PR.

**Pull request merged**